jerry-core/vm/opcodes.c

```diff
@@ -150,11 +150,6 @@ vm_op_delete_prop (ecma_value_t object, /**< base object */
                    ecma_value_t property, /**< property name */
                    bool is_strict) /**< strict mode */
 {
-  if (ecma_is_value_undefined (object))
```

Comment:
I think it cannot be removed, however it can be guarded to be present only in ES2015 profile.

Reply:
I found that this code was used in the following cases:
```js
delete 'test';
delete 0;
```
And return value of those cases are `true` in both es5.1 and es2015.

I'm wondering the issue would requiring a bytecode generation minor refactoring. Working on it.